library/cartridge_apply_app_config.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+
+import json
+import os
+import zipfile
+
+from ansible.module_utils.helpers import Helpers as helpers
+from ansible.module_utils.urls import open_url, basic_auth_header
+
+argument_spec = {
+    'local_config_path': {'required': True, 'type': 'str'},
+    'remote_dir': {'required': False, 'type': 'str', 'default': '/tmp/'},
+    'upload_endpoint': {'required': False, 'type': 'str'},
+    'console_sock': {'required': False, 'type': 'str'},
+    'http_port': {'required': False, 'type': 'int'},
+    'cluster_cookie': {'required': False, 'type': 'str'},
+    'tdg_token': {'required': False, 'type': 'str'},
+}
+
+LUA_ENDPOINT = 'lua'
+HTTP_ENDPOINT = 'http'
+TDG_ENDPOINT = 'tdg'
+
+PURE_DIR_MODE = 'pure'
+ZIP_DIR_MODE = 'zip'
+
+
+def patch_file_clusterwide(control_console, file_path):
+    file_ext = os.path.splitext(file_path)[1]
+    assert file_ext in ['.yml', '.yaml'], "Impossible to use '%s' file in patch clusterwide function" % file_ext
+
+    new_sections, err = helpers.read_yaml_file(control_console, file_path)
+    assert err is None, err
+
+    old_sections, err = helpers.get_clusterwide_config(control_console)
+    assert err is None, err
+
+    for section, value in old_sections.items():
+        new_sections[section] = new_sections.get(section)
+
+    changed, err = helpers.patch_clusterwide_config(control_console, new_sections)
+    assert err is None, err
+    return changed
+
+
+def send_on_http(http_port, headers, config_path):
+    url = 'http://127.0.0.1:{port}/admin/config'.format(port=http_port)
+    headers.update({
+        'Content-Length': os.stat(config_path).st_size
+    })
+
+    with open(config_path, 'rb') as data:
+        try:
+            resp = open_url(url, method='PUT', headers=headers, data=data)
+        except Exception as e:
+            assert hasattr(e, 'code'), str(e)
+            resp = e
+
+    if not 200 <= resp.code <= 299:
+        error = ''
+        try:
+            error = resp.read()
+            error = json.loads(error)
+            if "str" in error:
+                error = error["str"]
+        except Exception:
+            pass
+
+        raise AssertionError("Status: {}. Body: '{}'".format(resp.code, error))
+
+    return True
+
+
+def apply_tdg_config(console_sock, path):
+    control_console = helpers.get_control_console(console_sock)
+    _, err = control_console.eval_res_err('''
+        return admin.upload_config_api(...)
+    ''', path)
+    assert err is None, err
```

Comment:
Assert?

Reply:
Why not? It will be catched by module try-catch